### DISCUSSION TITLE
Setting to remove all pushes when stream ends

### DISCUSSION THREAD

**Discussion by irlkitcom** [OvenMediaLabs/OvenMediaEngine#1601] [Ideas]
It would be nice if all pushes could be removed when the incoming stream ends. Relying on Admission Webhooks seems to be futile considering this issue: https://github.com/AirenSoft/OvenMediaEngine/issues/1570

**Comment by bchah** [comment]
Interesting, I like that the push remains defined and can be stopped/started for the lifecycle of the application. And my app code depends on it 😅

But I do support the idea of adding some sort of keepalive parameter at the Server.xml level to the PushPublisher application.

e.g.

```
<Applications>
  <Application>
     ...
    <Publishers>
      ...
      <SRTPush>
       <KeepAlive>true</KeepAlive>
      </SRTPush>
    </Publishers>
  </Application>
</Applications>

```

In the meantime, a periodic GET call to the stats API will reveal which streams are live, followed by the appropriate DELETE calls to the Push API. It's a bit more work but gives more control over security too.

**Reply by irlkitcom** [reply]
I understand it's usefulness for certain applications like yours which is why I offered the idea as it's own setting, but for my application it would mean adding complexity of storing the unique IDs of every push and making sure to remove them if a stream has gone offline. It could end up being out of sync and causing issues.

**Reply by irlkitcom** [reply]
Are bounties allowed here? I'd be willing to offer a payment for someone who can create a PR for this feature.
